### PR TITLE
[BE][Easy]: Apply RUF019: remove duplicate checks for dict access

### PR DESCRIPTION
**Pull request by Skylion007**
Applies RUF019 nightly preview rule to the codebase

### CODE REVIEW

File: test/optim/test_optim.py [hunk 1]
```diff
@@ -851,7 +851,7 @@ def _test_foreach_memory(self, optimizer_pairs_with_flags):
             st_max_mem, mt_max_mem = max_mems
             intermediate_size = nparams * param.nelement() * param.element_size()
             nintermediates = 1  # we expect a budget of 1 intermediate most of the time
-            if (('capturable' in kwargs_with_flags and kwargs_with_flags['capturable']) or
+            if ((kwargs_with_flags.get('capturable')) or
```
Comment (janeyx99) [inline test/optim/test_optim.py:854]: looks like there’s an extra set of () here

**Review by janeyx99 — COMMENTED** [review]
looks fine, though is there a lint runner config update that would enforce this for the future?

or does the fact that RUF019 is still in preview prevent us?

**Comment by Skylion007** [comment]
It still being in preview does make it rather annoying to enforce as a lint yet.

**Review by mikaylagawarecki — APPROVED** [review]
Thanks!

**Pull request closed without merging**